Sources/TuistKit/Generator/GeneratorModelLoader.swift

```diff
@@ -285,15 +285,24 @@ extension TuistGenerator.InfoPlist.Value {
 }
 
 extension TuistGenerator.Settings {
+    typealias BuildConfigurationTuple = (TuistGenerator.BuildConfiguration, TuistGenerator.Configuration?)
```

Comment:
Looking at the code without having looked into this logic in a while, I find it confusing having to types named `BuildConfiguration` and `Configuration`. What's the difference? Could we use a different naming convention to clarify this?

Reply:
Indeed hence why I was questioning adding yet a 3rd type 😀

- `BuildConfiguration` is the identifier of the configuration so to speak (e.g. "Beta"), the reason this is more elaborate than just a `String` is due to requiring the knowledge of the variant to apply the appropriate default settings.
- `Configuration` is the build settings + xcconfig file reference

Reply:
I agree, but I can't think of a better name. Maybe some header documentation on the types would lessen confusion when developing features inside of Tuist?